### ISSUE TITLE
Deleting GraphNodes when handling `delete_nodes_request` causes connection visuals to leak

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#36716]
Godot 3.2
Windows 10 64 bits

I followed the way Godot's graph editors implement the `delete_nodes_request`:

```gdscript
func _on_GraphEdit_delete_nodes_request():
	var to_delete = []
	
	for i in _graph_edit.get_child_count():
		var node = _graph_edit.get_child(i)
		if node is GraphNode:
			if node.selected:
				to_delete.append(node)
	
	for node in to_delete:
		node.free() # queue_free also works
```

It properly removes the node, however it causes connections to remain visible, until the graph gets panned around or redrawn (which will then make them disappear):

![image](https://user-images.githubusercontent.com/1311555/75635214-2ea2b700-5c0c-11ea-93d8-e87c43a69ff5.png)

When redrawing, an error is logged once:
```
ERROR: Node not found: B.
   At: scene/main/node.cpp:1381
```

Note: I think this never happens in Godot's graph-based editors, because those actually rebuild the entire graph each time a node is deleted... which is very far from the way I implement mine.
This happens even if I use `queue_free()`.

Test project: 
[GraphEditDeleteRequest.zip](https://github.com/godotengine/godot/files/4272569/GraphEditDeleteRequest.zip)

If it's required to remove those connections manually, it would be really nice to have a function to do so.
Edit: indeed, it must be done manually.
```cpp
void remove_connections_from_and_to(GraphEdit *graph_edit, StringName node_name) {
	// Get copy of connection list
	List<GraphEdit::Connection> connections;
	graph_edit->get_connection_list(&connections);

	for (List<GraphEdit::Connection>::Element *E = connections.front(); E; E = E->next()) {
		const GraphEdit::Connection &con = E->get();
		if (con.from == node_name || con.to == node_name) {
			graph_edit->disconnect_node(con.from, con.from_port, con.to, con.to_port);
		}
	}
}
```

**Comment by YuriSizov** [comment]
Still reproduceable in 4.2 beta 6. cc @Geometror

**Comment by Geometror** [comment]
This is fixed in #86158.
The following actually works, but is hacky and shouldn't be used:
```
for node in to_delete:
		node.free() # queue_free also works
```
as a workaround.
